modules/photon/photon.js

```diff
@@ -1,54 +1,60 @@
+( function() {
+	function recalculate() {
+		if ( this.complete ) {
+			// Support for lazy loading: if there is a lazy-src attribute and it's value
+			// is not the same as the current src we should wait until the image load event
+			var lazySrc = this.getAttribute('data-lazy-src');
+			if ( lazySrc && this.src !== lazySrc ) {
+				this.addEventListener( 'load', recalculate );
+				return;
+			}
+
+			// Copying CSS width/height into element attributes.
+			var width = this.width;
+			var height = this.height;
+			if ( width && width > 0 && height && height > 0 ) {
+				this.setAttribute('width', width);
+				this.setAttribute('height', height);
+
+				reset_for_retina( this );
+			}
+		}
+		else {
+			this.addEventListener( 'load', recalculate );
+			return;
+		}
+	}
 	/**
 	 * For images lacking explicit dimensions and needing them, try to add them.
 	 */
 	var restore_dims = function() {
-			$( 'img[data-recalc-dims]' ).each( function recalc() {
-				var $this = $( this );
-				if ( this.complete ) {
-					// Support for lazy loading: if there is a lazy-src
-					// attribute and it's value is not the same as the current src we
-					// should wait until the image load event
-					if ( $this.data( 'lazy-src' ) && $this.attr( 'src' ) !== $this.data( 'lazy-src' ) ) {
-						$this.load( recalc );
-						return;
-					}
-
-					var width = this.width,
-						height = this.height;
-
-					if ( width && width > 0 && height && height > 0 ) {
-						$this.attr( {
-							width: width,
-							height: height,
-						} );
+		var elements = document.querySelectorAll( 'img[data-recalc-dims]' );
+		for (var i = 0; i < elements.length; i++) {
+			recalculate.call( elements[i] );
+		}
+	},
 
-						reset_for_retina( this );
-					}
-				} else {
-					$this.load( recalc );
-				}
-			} );
-		},
-		/**
-		 * Modify given image's markup so that devicepx-jetpack.js will act on the image and it won't be reprocessed by this script.
-		 */
-		reset_for_retina = function( img ) {
-			$( img )
-				.removeAttr( 'data-recalc-dims' )
-				.removeAttr( 'scale' );
-		};
+	/**
+	 * Modify given image's markup so that devicepx-jetpack.js will act on the image and it won't be reprocessed by this script.
+	 */
+	reset_for_retina = function( img ) {
+		img.removeAttribute( 'data-recalc-dims' );
+		img.removeAttribute( 'scale' );
+	};
 
+	// Vanilla version of jQuery.ready()
+	var ready = function( fn ) {
+		if ( document.attachEvent ? document.readyState === 'complete' : document.readyState !== 'loading' ) {
+			fn();
+		} else {
+			document.addEventListener( 'DOMContentLoaded', fn );
+		}
+	};
 	/**
 	 * Check both when page loads, and when IS is triggered.
 	 */
-	$( document ).ready( restore_dims );
-
-	if ( 'on' in $.fn ) {
-		$( document.body ).on( 'post-load', restore_dims );
-	} else {
-		$( document ).delegate( 'body', 'post-load', restore_dims );
-	}
-} )( jQuery );
+	ready( restore_dims );
+	document.body.addEventListener( 'post-load', restore_dims );
```

Comment:
Here's another, what I think is more simple/readable approach for attaching onload event:

https://github.com/Automattic/jetpack/blob/201ec52be66261a364f0805a6e44d25269d84793/extensions/blocks/tiled-gallery/view.js#L57-L64

Note also window/document protector in case the script runs in context which doesn't have those (e.g. some test environments).

Reply:
Just realised and got pointed out by @brbrr that these are different events (`DOMContentLoaded` and `post-load`). I'm not familiar of the difference. :-)

I was pointing out about code readability, not about which event to use. I felt that instead of:

```
ready( restore_dims );
document.body.addEventListener( 'post-load', restore_dims );
```

having the same as `if() { } else { }` format was clearer, but again I'm not perfectly sure if these examples do the same thing. :-)

I'd say don't worry and if it works, leave it as it is! 😅

Reply:
`post-load` is an infinite scroll custom event, so I think we should use both 🤷‍♂️